File: python-algorithms/lists/linked_list_problems.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def reverse_linked_list(self, head):
        prev = None
        current = head

        while current:
            tmp = current.next
            current.next = prev
            prev = current
            current = tmp
        return prev


def printList(node):
    current = node
    while current:
        print(current.val)
        current = current.next


head = ListNode(1)
head.next = ListNode(2)
head.next = ListNode(3)
solution = Solution()
printList(solution.reverse_linked_list(head))

File: java-algorithms/src/main/java/org/mddarr/lists/LinkedListProblems.java
package org.mddarr.lists;

public class LinkedListProblems {
    public static void main(String[] args){
//        ListNode head = new ListNode(1);
//        head.next = new ListNode(2);
//        head.next.next = new ListNode(3);
//        head.next.next.next = new ListNode(4);
//        Solution solution = new Solution();
//        ListNode reversedList = solution.reverseList(head);
//        solution.reverseList(reversedList);


//        Solution solution = new Solution();
//        ListNode cyclicHead = new ListNode(1);
//        cyclicHead.next = new ListNode(2);
//        cyclicHead.next.next = new ListNode(3);
//        cyclicHead.next.next.next = new ListNode(1);
//        boolean iscyclic = solution.hasCycle(cyclicHead);
//        System.out.println("The linked list is cyclic " + iscyclic);
        Solution solution = new Solution();
        ListNode lhead = new ListNode(1);
//        lhead.next = new ListNode(2);
//        lhead.next.next = new ListNode(3);
//        lhead.next.next.next = new ListNode(4);
//        lhead.next.next.next.next = new ListNode(5);
        ListNode newHead = solution.rotateRight(lhead,1);

        solution.printList(newHead);

    }

}
